operations/fetch.go

```diff
@@ -189,75 +180,77 @@ func fetchSource(ac, rc *legacyClient, rootPath, taskId, token string, noPatch b
 }
 
 type cloneOptions struct {
-	token      string
-	owner      string
-	repository string
-	revision   string
-	rootDir    string
-	branch     string
-	depth      uint
+	repo     string
+	revision string
+	rootDir  string
+	branch   string
+	depth    uint
 }
 
 func clone(opts cloneOptions) error {
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
-	defer cancel()
-
-	var url string
-	if opts.token != "" {
-		url = fmt.Sprintf("https://%s@github.com/%s/%s.git", opts.token, opts.owner, opts.repository)
-	} else {
-		url = fmt.Sprintf("git@github.com:%v/%v.git", opts.owner, opts.repository)
-	}
-	cloneOpts := git.CloneOptions{
-		URL:      url,
-		Depth:    int(opts.depth),
-		Progress: os.Stdout,
+	// clone the repo first
```

Comment:
Should we move this to thirdparty/git?

Reply:
We could, though this was in this file beforehand.

Reply:
We probably should but it can be another ticket. The code is duplicated ish with git.get_project